extension/parquet/parquet-extension.cpp

```diff
@@ -44,6 +45,8 @@ struct ParquetReadBindData : public TableFunctionData {
 	atomic<idx_t> cur_file;
 	vector<string> names;
 	vector<LogicalType> types;
+	// Indicates that the file from the initial_reader is no longer present in files
+	bool should_flush_initial_reader = false;
```

Comment:
Can't we remove the initial reader entirely instead of having this boolean value indicating it has been invalidated?

Reply:
yes, it required a little rewrite since the initial_reader was used for reading metadata in the the ParquetScanStats function. Added some extra test cases for this.